### PR TITLE
Add initial locale sharing proposal

### PR DESCRIPTION
**Pull request by cutandpastey**
See markdown document diffs for details:

https://github.com/festicket/front-end-meta/blob/jp-locale/proposals/locale.md

### CODE REVIEW

File: proposals/locale.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  }
+</Locale.Context>
+```
+
+### Changing Locale
+
+The simplest way to reload an entire application is to reload the whole page.
+As the currently selected locale is stored in the API we can do this with confidence.
+We can negate some of the performance implications of this approach by leveraging
+Service Worker caching so page reloads can "feel" instant.
```
Comment (dbbk) [inline proposals/locale.md]: I agree reloading is the best way to go. Though I don't think adding a Service Worker will be strictly necessary here, as long as our assets are served with long or `immutable` cache headers, the reload will be fast. The only bottleneck would be the server-rendering of the HTML response, which would also be alleviated once we get to caching HTML responses on the CDN in the future.

File: proposals/locale.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+- We need to know what the currently selected locale is.
+- We need to know what to do once the locale is changed
+
+
+## Solution:
+### Determining the current locale
+At festicket we rely on the url to determine the currently selected locale.
+Whilst this is not perfect this **is** a business requirement.
+On an application level we will need a server-side middleware that can query the festicket API
```
Comment (dbbk) [inline proposals/locale.md]: Will this need to be a standalone API endpoint or can we merge it with the 'current user details' request?

Comment (cutandpastey) [reply proposals/locale.md]: I don't actually know. Depends whats faster, ideally we want to defer loading the user until we are in the client context. I suppose we could add it to the user and then just rewrite the url on the client once we get the user object back and there is a mismatch. The API will provide translated content either way.

 /cc @mjjared

Comment (dbbk) [reply proposals/locale.md]: Ah yes

File: proposals/locale.md (hunk 1 above)
Comment (reyrodrigez) [reply proposals/locale.md]: with caching on the service worker level we can enable the offline feature which could be beneficial for the magazine.

**Pull request merged**